### ISSUE TITLE
plot show nothing and get stuck

### ISSUE THREAD

**Issue by LuhengStat** [jorgenschaefer/elpy#1337]
# Summary

When I use the code below to plot a figure, there has nothing show up and get stuck.

import matplotlib.pyplot as plt
plt.plot([1,2,3,4])
plt.ylabel('some numbers')
plt.show()

However, these codes can be run in terminal

# Steps to reproduce

# My configuration
## Result of `(elpy-config)`
```
Elpy Configuration

Virtualenv........: None
RPC Python........: 3.6.4 (/anaconda3/bin/python)
Interactive Python: python (/anaconda3/bin/python)
Emacs.............: 25.3.1
Elpy..............: 1.19.0
Jedi..............: 0.11.1
Rope..............: 0.10.7
Autopep8..........: 1.3.5
Yapf..............: 0.21.0
Syntax checker....: Not found (flake8)

You have not activated a virtual env. While Elpy supports this, it is
often a good idea to work inside a virtual env. You can use M-x
pyvenv-activate or M-x pyvenv-workon to activate a virtual env.

The directory ~/.local/bin/ is not in your PATH. As there is no active
virtualenv, installing Python packages locally will place executables
in that directory, so Emacs won't find them. If you are missing some
commands, do add this directory to your PATH -- and then do
`elpy-rpc-restart'.

The configured syntax checker could not be found. Elpy uses this
program to provide syntax checks of your programs, so you might want
to install one. Elpy by default uses flake8.

[run] python -m pip install --user flake8

Options

`Raised' text indicates buttons; type RET or click mouse-1 on a button
to invoke its action.  Invoke [+] to expand a group, and [-] to
collapse an expanded group.  Invoke the [Group], [Face], and [Option]
buttons below to edit that item in another window.

[+]-- Group Elpy
[+]-- Group Python
[+]-- Group Virtual Environments (Pyvenv)
[+]-- Group Completion (Company)
[+]-- Group Call Signatures (ElDoc)
[+]-- Group Inline Errors (Flymake)
[+]-- Group Snippets (YASnippet)
[+]-- Group Directory Grep (rgrep)
[+]-- Group Search as You Type (ido)
[+]-- Group Django extension
[+]-- Group Autodoc extension

```

## Elpy configuration in my init.el
```

(package-initialize)
(elpy-enable)
(add-hook 'elpy-mode-hook (lambda () (highlight-indentation-mode -1)))

(add-hook 'inferior-python-mode-hook 'visual-line-mode)
(define-key elpy-mode-map (kbd "C-x C-n") 'elpy-shell-send-group-and-step)

```

**Comment by galaunay** [comment]
Unfortunately, I was not able to reproduce this.

My first guess would be different matplotlib configurations (for some reason...).
Is there any difference in `plt.rcParams` between the terminal and elpy ?
More specifically the backend ?

**Comment by LuhengStat** [comment]
After I changed to jupyter, and the problem has been solved. cheer. below is the settings.
Sorry for open two issues for these.

(package-initialize)
(elpy-enable)
(add-hook 'elpy-mode-hook (lambda () (highlight-indentation-mode -1)))

(add-hook 'inferior-python-mode-hook 'visual-line-mode)
(define-key elpy-mode-map (kbd "C-x C-n") 'elpy-shell-send-group-and-step)

(setq python-shell-prompt-detect-failure-warning nil)
(setq python-shell-completion-native-enable nil) 
(setq python-shell-interpreter "jupyter"
      python-shell-interpreter-args "console --simple-prompt")

If the figure shows not very clear, just set the matplotlib backend by osx.

%matplotlib
Using matplotlib backend: MacOSX
and here gives the solution to set the default backend with osx.
https://stackoverflow.com/questions/21176731/automatically-run-matplotlib-inline-in-ipython-notebook
